homeassistant/components/rest/data.py

```diff
@@ -71,22 +71,26 @@ def set_url(self, url: str) -> None:
 
     def data_without_xml(self) -> str | None:
         """If the data is an XML string, convert it to a JSON string."""
-        _LOGGER.debug("Data fetched from resource: %s", self.data)
-        if (
-            (value := self.data) is not None
-            # If the http request failed, headers will be None
-            and (headers := self.headers) is not None
-            and (content_type := headers.get("content-type"))
-            and content_type.startswith(XML_MIME_TYPES)
-        ):
-            try:
-                value = json_dumps(xmltodict.parse(value))
-            except ExpatError:
-                _LOGGER.warning(
-                    "REST xml result could not be parsed and converted to JSON"
-                )
-            else:
-                _LOGGER.debug("JSON converted from XML: %s", value)
+        _LOGGER.debug("Data fetched from resource (%s): %s", self._resource, self.data)
+
+        if (value := self.data) is not None:
+            xmlContentIndicator = value.lower().startswith(XML_CONTENT_INDICATORS)
+            xmlFileExtension = self._resource.lower().endswith(XML_FILENAME_EXTENSIONS)
+            xmlContentType = (
+                (headers := self.headers) is not None
+                and (content_type := headers.get("content-type"))
+                and content_type.lower().startswith(XML_MIME_TYPES)
+            )
```

Comment:
This is nit but the diff will be smaller (no extra nesting level) and the condition check has opportunity to evaluate only _some_ conditions, not all of them.
```suggestion
        if (
            (value := self.data) is not None
            and (
                value.lower().startswith(XML_CONTENT_INDICATORS)
                or self._resource.lower().endswith(XML_FILENAME_EXTENSIONS)
                or (
                    (headers := self.headers) is not None
                    and (content_type := headers.get("content-type"))
                    and content_type.lower().startswith(XML_MIME_TYPES)
                )
            )):
```

Reply:
That's fair, i need to rebase this due to upstream changes anyway so I'll incorporate this and the upstream changes

Reply:
I set the PR to draft since the rebase does not seem to be done. Please click the "Ready for review" button once you've done the rebase 👍